File: examples/colors.cpp
// =============================================================================
// C++ Color Formats Example
// nvim-colorpicker detects and can replace all these formats
// =============================================================================

#include <QColor>
#include <cstdint>

// Qt QColor with integer RGB (0-255)
namespace QtColors {
    const QColor Primary = QColor(98, 0, 238);
    const QColor Secondary = QColor(3, 218, 198);
    const QColor Background = QColor(18, 18, 18);
    const QColor Surface = QColor(30, 30, 30);
    const QColor Error = QColor(207, 102, 121);
    const QColor Success = QColor(102, 187, 106);
}

// Qt QColor with alpha
namespace TransparentColors {
    const QColor Overlay = QColor(0, 0, 0, 128);
    const QColor Highlight = QColor(98, 0, 238, 64);
    const QColor Glass = QColor(255, 255, 255, 26);
    const QColor Shadow = QColor(0, 0, 0, 153);
}

// Qt QColor with float RGB (0.0-1.0)
namespace FloatColors {
    const QColor Primary = QColor::fromRgbF(0.384, 0.000, 0.933);
    const QColor Secondary = QColor::fromRgbF(0.012, 0.855, 0.776);
    const QColor Background = QColor::fromRgbF(0.071, 0.071, 0.071);
}

// Qt QColor with float RGBA
namespace FloatAlphaColors {
    const QColor Overlay = QColor::fromRgbF(0.000, 0.000, 0.000, 0.50);
    const QColor Highlight = QColor::fromRgbF(0.384, 0.000, 0.933, 0.25);
    const QColor Subtle = QColor::fromRgbF(1.000, 1.000, 1.000, 0.10);
}

// Numeric hex format (0xRRGGBB)
namespace HexColors {
    constexpr uint32_t RED = 0xFF5722;
    constexpr uint32_t PINK = 0xE91E63;
    constexpr uint32_t PURPLE = 0x9C27B0;
    constexpr uint32_t BLUE = 0x2196F3;
    constexpr uint32_t CYAN = 0x00BCD4;
    constexpr uint32_t GREEN = 0x4CAF50;
    constexpr uint32_t YELLOW = 0xFFEB3B;
    constexpr uint32_t ORANGE = 0xFF9800;
    constexpr uint32_t WHITE = 0xFFFFFF;
    constexpr uint32_t BLACK = 0x000000;
}

// ARGB format (0xAARRGGBB)
namespace ArgbColors {
    constexpr uint32_t SOLID_RED = 0xFFFF0000;
    constexpr uint32_t SOLID_GREEN = 0xFF00FF00;
    constexpr uint32_t SOLID_BLUE = 0xFF0000FF;
    constexpr uint32_t SEMI_BLACK = 0x80000000;
    constexpr uint32_t TRANSPARENT = 0x00000000;
}

// Struct initializer style
struct Color {
    uint8_t r, g, b, a;
};

namespace StructColors {
    constexpr Color Red = {255, 0, 0, 255};
    constexpr Color Green = {0, 255, 0, 255};
    constexpr Color Blue = {0, 0, 255, 255};
    constexpr Color Yellow = {255, 255, 0, 255};
    constexpr Color Cyan = {0, 255, 255, 255};
    constexpr Color Magenta = {255, 0, 255, 255};
    constexpr Color White = {255, 255, 255, 255};
    constexpr Color Black = {0, 0, 0, 255};
    constexpr Color SemiTransparent = {0, 0, 0, 128};
}

// Float struct style (0.0-1.0)
struct ColorF {
    float r, g, b, a;
};

namespace FloatStructColors {
    constexpr ColorF Primary = {0.384f, 0.000f, 0.933f, 1.00f};
    constexpr ColorF Secondary = {0.012f, 0.855f, 0.776f, 1.00f};
    constexpr ColorF Background = {0.071f, 0.071f, 0.071f, 1.00f};
    constexpr ColorF Overlay = {0.000f, 0.000f, 0.000f, 0.50f};
}

// Dark theme
namespace DarkTheme {
    const QColor Background = QColor(26, 26, 46);
    const QColor Surface = QColor(22, 33, 62);
    const QColor Primary = QColor(15, 52, 96);
    const QColor Accent = QColor(233, 69, 96);
    const QColor Text = QColor(234, 234, 234);
    const QColor TextMuted = QColor(170, 175, 191, 170);
}

// Game colors
namespace GamePalette {
    constexpr Color Player = {79, 195, 247, 255};
    constexpr Color Enemy = {239, 83, 80, 255};
    constexpr Color Collectible = {255, 213, 79, 255};
    constexpr Color Obstacle = {120, 144, 156, 255};
    constexpr Color Health = {102, 187, 106, 255};
    constexpr Color Mana = {171, 71, 188, 255};
}

File: examples/colors.rs
// =============================================================================
// Rust Color Formats Example
// nvim-colorpicker detects and can replace all these formats
// =============================================================================

use bevy::prelude::*;

// Bevy srgb format (floats 0.0-1.0)
pub mod bevy_colors {
    use super::*;

    pub const PRIMARY: Color = Color::srgb(0.384, 0.000, 0.933);
    pub const SECONDARY: Color = Color::srgb(0.012, 0.855, 0.776);
    pub const BACKGROUND: Color = Color::srgb(0.071, 0.071, 0.071);
    pub const SURFACE: Color = Color::srgb(0.118, 0.118, 0.118);
    pub const ERROR: Color = Color::srgb(0.812, 0.400, 0.475);
    pub const SUCCESS: Color = Color::srgb(0.400, 0.733, 0.416);
}

// Bevy srgba format (with alpha)
pub mod transparent_colors {
    use super::*;

    pub const OVERLAY: Color = Color::srgba(0.000, 0.000, 0.000, 0.50);
    pub const HIGHLIGHT: Color = Color::srgba(0.384, 0.000, 0.933, 0.25);
    pub const GLASS: Color = Color::srgba(1.000, 1.000, 1.000, 0.10);
    pub const SHADOW: Color = Color::srgba(0.000, 0.000, 0.000, 0.60);
    pub const FADE: Color = Color::srgba(0.071, 0.071, 0.071, 0.00);
}

// Bevy hex format
pub mod hex_colors {
    use super::*;

    pub const RED: Color = Color::hex("#FF5722").unwrap();
    pub const PINK: Color = Color::hex("#E91E63").unwrap();
    pub const PURPLE: Color = Color::hex("#9C27B0").unwrap();
    pub const BLUE: Color = Color::hex("#2196F3").unwrap();
    pub const CYAN: Color = Color::hex("#00BCD4").unwrap();
    pub const GREEN: Color = Color::hex("#4CAF50").unwrap();
    pub const YELLOW: Color = Color::hex("#FFEB3B").unwrap();
    pub const ORANGE: Color = Color::hex("#FF9800").unwrap();

    // Without # prefix
    pub const WHITE: Color = Color::hex("FFFFFF").unwrap();
    pub const BLACK: Color = Color::hex("000000").unwrap();

    // With alpha
    pub const SEMI_RED: Color = Color::hex("#80FF5722").unwrap();
}

// Dark theme
pub mod dark_theme {
    use super::*;

    pub const BACKGROUND: Color = Color::srgb(0.102, 0.102, 0.180);
    pub const SURFACE: Color = Color::srgb(0.086, 0.129, 0.243);
    pub const PRIMARY: Color = Color::srgb(0.059, 0.204, 0.376);
    pub const ACCENT: Color = Color::srgb(0.914, 0.271, 0.376);
    pub const TEXT: Color = Color::srgb(0.918, 0.918, 0.918);
    pub const TEXT_MUTED: Color = Color::srgba(0.667, 0.686, 0.749, 0.667);
}

// Game colors
pub mod game_palette {
    use super::*;

    pub const PLAYER: Color = Color::srgb(0.310, 0.765, 0.969);
    pub const ENEMY: Color = Color::srgb(0.937, 0.325, 0.314);
    pub const COLLECTIBLE: Color = Color::srgb(1.000, 0.835, 0.310);
    pub const OBSTACLE: Color = Color::srgb(0.471, 0.565, 0.612);
    pub const HEALTH: Color = Color::srgb(0.400, 0.733, 0.416);
    pub const MANA: Color = Color::srgb(0.671, 0.278, 0.737);
}

// Macroquad style (if using macroquad crate)
#[cfg(feature = "macroquad")]
pub mod macroquad_colors {
    use macroquad::prelude::*;

    pub const RED: Color = Color::new(1.00, 0.00, 0.00, 1.00);
    pub const GREEN: Color = Color::new(0.00, 1.00, 0.00, 1.00);
    pub const BLUE: Color = Color::new(0.00, 0.00, 1.00, 1.00);
    pub const YELLOW: Color = Color::new(1.00, 1.00, 0.00, 1.00);
    pub const CYAN: Color = Color::new(0.00, 1.00, 1.00, 1.00);
    pub const MAGENTA: Color = Color::new(1.00, 0.00, 1.00, 1.00);

    // Semi-transparent
    pub const GLASS: Color = Color::new(0.80, 0.90, 1.00, 0.20);
    pub const SHADOW: Color = Color::new(0.00, 0.00, 0.00, 0.50);
}

// Gradient colors
pub fn sunset_gradient() -> Vec<Color> {
    vec![
        Color::srgb(1.000, 0.341, 0.133),
        Color::srgb(1.000, 0.596, 0.000),
        Color::srgb(1.000, 0.922, 0.231),
    ]
}

pub fn ocean_gradient() -> Vec<Color> {
    vec![
        Color::srgb(0.000, 0.737, 0.831),
        Color::srgb(0.129, 0.588, 0.953),
        Color::srgb(0.247, 0.318, 0.710),
    ]
}
